src/components/navbar.js

```diff
@@ -0,0 +1,214 @@
+import { useEffect, useState } from "react";
+import {
+  BrowserRouter as Router,
+  Switch,
+  Route,
+  Link,
+  useHistory,
+  useRouteMatch,
+} from "react-router-dom";
+
+export default function Navbar(props) {
+  useEffect(() => {
+    let element = document.getElementById("main-navbar");
+    if (props.show) {
+      element.classList.add("sticky-header");
+    } else {
+      element.classList.remove("sticky-header");
+    }
+  }, [props.show]);
+
+  //const [currentLocation, setCurrentLocation] = useState("");
+  const [currentWindowLocation, setCurrentWindowLocation] = useState(
+    window.location.href
+  );
+
+  function handleCurrentLocation(e) {
+    //setCurrentLocation(e);
+    setCurrentWindowLocation(window.location.href);
+  }
+
+  useEffect(() => {
+    //console.log(currentLocation);
+    //console.log("currentWindowLocation", currentWindowLocation);
+    const menuItems = document.getElementsByClassName("menu-items");
+    //console.log(menuitems)
+    let menuItemFound = false;
+    for (const item of menuItems) {
+      //console.log('item.href',item.href)
+      const found = currentWindowLocation === item.href;
+
+      if (found) {
+        menuItemFound = found;
+        //console.log(item.innerHTML, 'found')
+        item.classList.add("active-page");
+      } else {
+        item.classList.remove("active-page");
+      }
+    }
+
+    // drop-down sub-menu items
+    if (!menuItemFound) {
+      const subMenuItems = document.getElementsByClassName(
+        "submenu-anchor-item"
+      );
+      const dropDownMenuItem = document.getElementById("dropDownMenuItem");
+      for (const subItem of subMenuItems) {
+        const found = currentWindowLocation === subItem.href;
+        //console.log("subItem.href", subItem.href);
+        if (found) {
+          menuItemFound = found;
+          //console.log(subItem.innerHTML, "found");
+          subItem.classList.add("sub-active-page");
+          dropDownMenuItem.classList.add("active-page");
+        } else {
+          subItem.classList.remove("sub-active-page");
+        }
+      }
+    }
+  }, [currentWindowLocation]);
+
+  const handleClick = (e) => {
+    //console.log(e.target);
+    const clickedItem = e.target.innerHTML;
+    const regex = /\b\s\b/;
+    const idCorrectedItem = clickedItem.replace(regex, "_").toLowerCase();
+    const element = document.getElementById(idCorrectedItem);
+
+    switch (clickedItem) {
+      case "Home":
+        window.scroll(0, 0);
+        break;
+
+      case "About":
+        window.scroll(0, element.offsetTop - 260);
```

Comment:
instead of giving hardcoded numbers to offset, I would put an element as an anchor and give it negative margin top and just use anchor link instead of this script
```
<div class="anchor" id="top"></div>
You can then position the anchor an offset higher or lower than where it actually appears on the page, by making it a block element and relatively positioning it. -250px will position the anchor up 250px

anchor {
    display: block;
    position: relative;
    top: -250px;
    visibility: hidden;
}

Reply:
Can you clarify how to use this anchor tag please?

Reply:
probably better to do it in a video call